### PR TITLE
Extract public URI in akka-http routing

### PR DESCRIPTION
**Pull request by warrd**
We need to set the correct `host` and `scheme`, and remove `port`, otherwise we get links which include  `https://localhost:8888/...`

### CODE REVIEW

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala [hunk 1]
```diff
@@ -221,6 +221,14 @@ class Router(elasticClient: ElasticClient,
       }
     }
 
+  def extractPublicUri =
+    extractUri.map { uri =>
+      uri
+        .withHost(apiConfig.host)
+        .withPort(0) // Port 0 is used here to indicate to akka no explicit port
+        .withScheme(apiConfig.scheme)
+    }
```
Comment (alexwlchan) [inline api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala]: Perhaps worth including the comment from the PR description explaining why we're doing this, and what the consequences of not doing it are? With examples of before/after URIs?

Comment (warrd) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala]: :+1: done

Comment (alexwlchan) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala]: What should the URIs look like after you've done this? Also I liked the comment about `withPort(0)` being a way to tell Akka not to set an explicit port.

Comment (warrd) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala]: Have updated, what about now?

**Pull request merged**